diff --git a/scripts/build/libc/glibc-eglibc.sh-common b/scripts/build/libc/glibc-eglibc.sh-common
--- a/scripts/build/libc/glibc-eglibc.sh-common
+++ b/scripts/build/libc/glibc-eglibc.sh-common
@@ -460,7 +460,7 @@
             # EGLIBC doesn't have a install-{pdf.html}, and older
             # versions leave the manuals in the source directory
             CT_DoExecLog ALL mkdir -p ${CT_PREFIX_DIR}/share/doc
-            if [ ${CT_LIBC_EGLIBC_2_16_or_later} = "y" ]; then
+            if [ "${CT_LIBC_EGLIBC_2_16_or_later}" = "y" ]; then
                 CT_DoExecLog ALL cp -av manual/*.pdf manual/libc \
                     ${CT_PREFIX_DIR}/share/doc
             else
